Tribler/Policies/BoostingManager.py

```diff
@@ -0,0 +1,500 @@
+            elif source.startswith('http://') or source.startswith('https://'):
+                self.boosting_sources[source] = RSSFeedSource(*args)
+            elif len(source) == 20:
+                self.boosting_sources[source] = ChannelSource(*args)
+            else:
+                self._logger.error("Cannot add unknown source %s", source)
+                return
+
+            self._logger.info("Added source %s", source)
+        else:
+            self._logger.info("Already have source %s", source)
+
+    def remove_source(self, source_key):
+        """
+        remove source by stop the downloading and remove its metainfo for all its swarms
+        """
+        if source_key in self.boosting_sources:
+            source = self.boosting_sources.pop(source_key)
+            source.kill_tasks()
+            self._logger.info("Removed source %s", source_key)
+
+            rm_torrents = [torrent for _, torrent in self.torrents.items()
+                           if torrent['source'] == source_to_string(source_key)]
+
+            for torrent in rm_torrents:
+                self.stop_download(torrent)
+                self.torrents.pop(torrent["metainfo"].get_infohash(), None)
+
+            self._logger.info("Torrents download stopped and removed")
+
+    def on_torrent_insert(self, source, infohash, torrent):
+        """
+        This function called when a source is finally determined. Fetch some torrents from it,
+        then insert it into our data
+        """
+
+        # Remember where we got this torrent from
+        self._logger.debug("remember torrent %s from %s", torrent, source_to_string(source))
+
+        torrent['source'] = source_to_string(source)
+
+        boost_source = self.boosting_sources.get(source, None)
+        if not boost_source:
+            self._logger.info("Dropping torrent insert from removed source: %s", repr(torrent))
+            return
+        elif boost_source.archive:
+            torrent['preload'] = True
+            torrent['prio'] = 100
+
+        # If duplicates exist, set is_duplicate to True, except for the one with the most seeders.
+        duplicates = [other for other in self.torrents.values() if compare_torrents(torrent, other)]
+        if duplicates:
+            duplicates += [torrent]
+            healthiest_torrent = max([(torrent['num_seeders'], torrent) for torrent in duplicates])[1]
+            for duplicate in duplicates:
+                is_duplicate = healthiest_torrent != duplicate
+                duplicate['is_duplicate'] = is_duplicate
+                if is_duplicate and duplicate.get('download', None):
+                    self.stop_download(duplicate)
+
+        self.torrents[infohash] = torrent
+
+    def on_torrent_notify(self, subject, change_type, infohash):
+        """
+        Notify us when we have new seeder/leecher value in torrent from tracker
+        """
+        if infohash not in self.torrents:
+            return
```

Comment:
This is called when we get notification of torrent updates, which is not exclusively used in credit mining